```
[ERROR] Failures: 
[ERROR]   UserApiOnPremTest.userMetadataOnPremTest:55 user-without-applications.json (with ignored fields) expected:<[\Q{"isPublic":\E[^,:\[\]{}]*\Q,"user":{"name":"Joe Developer","email":"dev@domail","nickname":"dev"},"tenants":{},"operator":["hostedOperator"]}\E]> but was:<[{"isPublic":false,"user":{"name":"Joe Developer","email":"dev@domail","nickname":"dev"},"tenants":{},"operator":["hostedSupporter","hostedOperator"]}]>
```

--- a/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/user/UserApiHandler.java
+++ b/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/user/UserApiHandler.java
@@ -136,6 +136,7 @@ public class UserApiHandler extends LoggingRequestHandler {
         List<Role> operatorRoles = roles.stream()
                 .filter(role -> role.definition().equals(RoleDefinition.hostedOperator) ||
                         role.definition().equals(RoleDefinition.hostedSupporter))
+                .sorted(Comparator.comparing(Role::definition))
                 .collect(Collectors.toList());
 
         Slime slime = new Slime();

--- a/controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/user/UserApiTest.java
+++ b/controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/user/UserApiTest.java
@@ -203,7 +203,7 @@ public class UserApiTest extends ControllerContainerCloudTest {
     public void userMetadataTest() {
         ContainerTester tester = new ContainerTester(container, responseFiles);
         ControllerTester controller = new ControllerTester(tester);
-        Set<Role> operator = Set.of(Role.hostedOperator());
+        Set<Role> operator = Set.of(Role.hostedOperator(), Role.hostedSupporter());
         User user = new User("dev@domail", "Joe Developer", "dev", null);
 
         tester.assertResponse(request("/api/user/v1/user")

--- a/controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/user/responses/user-without-applications.json
+++ b/controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/user/responses/user-without-applications.json
@@ -7,6 +7,7 @@
   },
   "tenants": {},
   "operator": [
-    "hostedOperator"
+    "hostedOperator",
+    "hostedSupporter"
   ]
 }

